File: server/routes/temp.js
const router = require('express').Router();
const auth = require('../middlewares/auth');

// Example of a guest-only temporary endpoint
router.post('/draft', auth, async (req, res) => {
  if (req.auth?.type !== 'guest') {
    return res.status(400).json({ message: 'Use guest token for this endpoint' });
  }

  // Handle temporary/preview work here (no permanent DB writes)
  return res.json({ ok: true, sessionId: req.auth.sessionId });
});

module.exports = router;

File: server_2/routes/temp.py
from fastapi import APIRouter, Depends, HTTPException, status

from database import get_database
from middlewares.auth import AuthContext, require_auth

router = APIRouter()


@router.post("/draft")
async def draft(auth: AuthContext = Depends(require_auth), db=Depends(get_database)):  # db dependency keeps parity for future use
    if auth.get("type") != "guest":
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST, detail="Use guest token for this endpoint"
        )
    return {"ok": True, "sessionId": auth.get("sessionId")}
